### PR TITLE
BOOKKEEPER-1009 Use multiple journals in bookie

### PR DESCRIPTION
**Pull request by govind-menon**
Mostly https://github.com/apache/bookkeeper/pull/71 with comments addressed

Will shortly add tests.

@merlimat  Would it be ok to close 71?

### CODE REVIEW

**Comment by govind-menon** [comment]
Failing tests are also on master. Tagging @eolivelli for review.

File: bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/Journal.java [hunk 1]
```diff
@@ -948,6 +953,8 @@ public void run() {
             LOG.error("I/O exception in Journal thread!", ioe);
         } catch (InterruptedException ie) {
             LOG.warn("Journal exits when shutting down", ie);
+        } catch (Throwable e) {
```
Comment (eolivelli) [inline bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/Journal.java]: @govind-menon do we need to catch Throwable ? do you have cases of unchecked exceptions, like OutOfMemory errors ?

**Comment by govind-menon** [comment]
@eolivelli  Done.

**Comment by govind-menon** [comment]
@eolivelli Pleaes l.et me know if there are other changes to be made. Thanks again for the review :)

**Comment by eolivelli** [comment]
@govind-menon for me is +1

ping @sijie  @jvrao for review and merge

**Comment by sijie** [comment]
@eolivelli checking

**Comment by govind-menon** [comment]
@sijie Thank you for the review and merge :)

**Comment by sijie** [comment]
Thank you @govind-menon and @merlimat !

Thank you @eolivelli for reviewing this one.

**Pull request closed without merging**